File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        occurance = {}
        for i, num in enumerate(nums):
            if num in occurance.keys():
                occurance[num].append(i)
            else:
                occurance[num] = [i]

        for i, num in enumerate(nums):
            if target - num == num:
                if len(occurance[num]) == 2:
                    return occurance[num]
            else:
                if (target - num) in occurance.keys():
                    return [i, occurance[target - num][0]]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
    i := 0
    j := len(numbers) - 1
    for i < j && numbers[i] + numbers[j] != target {
        for numbers[i] + numbers[j] > target && i < j {
            j--
        }
        for numbers[i] + numbers[j] < target && i < j {
            i++
        }
    }
    return []int{i+1,j+1}
}
